Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -1726,11 +1726,13 @@ void SoftwareGLContext::gl_bind_texture(GLenum target, GLuint texture)
     }
 
     auto it = m_allocated_textures.find(texture);
+    RefPtr<Texture> texture_object;
 
-    // The texture name does not exist
-    RETURN_WITH_ERROR_IF(it == m_allocated_textures.end(), GL_INVALID_VALUE);
-
-    auto texture_object = it->value;
+    // If there is not an allocated texture, we can keep texture_object null to both allocate and bind the texture with the passed in texture name.
```

Comment:
I'm not too sure about this comment, please tell me what you think about it :^)

Reply:
I would simplify this a bit, e.g.
> OpenGL 1.x supports texture names that were not previously generated by `glGenTextures`

Optionally a `FIXME` to support/restore the behavior as defined in newer versions?

The Xash3D mention should go in the commit message since that might be outdated quickly